docs/rfcs/9999-vpn-firefox-integration.md

```diff
@@ -0,0 +1,164 @@
+- The `allowedIp` for this activation is set to `10.124.0.0/20` so that only the Mullvad Exit Proxies are accessible on the VPN Network Device. Therefore, all other traffic will not use the VPN. 
+- A Mullvad DNS server will be set, so that the Mullvad proxies are resolvable. If a custom DNS is set, we need to set a Mullvad DNS as 2nd DNS server.
+- The Client UI does not show an activation. (or a new special UI state)
+> Q: Do we need a new UI state ? or like a "firefox is using the vpn?" 
+- If the Client activates, it will attempt to switch to `StateOn` and protect the whole device.
+- An Extension may request `{"t":"deactivate"}` only if the client is in `StateOnPartial` and it was the activator. The Client will then go to `StateOff`.
+
+The extension must then set the proxy to the `socks5` URL of the current `exitServer` OR, if a custom exit location is set, to the `socks5` proxy of that endpoint.
+
+#### State 3: Device VPN is On & Page VPN is Off
+
+If the full device is protected and the VPN integration is enabled, we will utilize a Socks5 Proxy running on localhost that will make sure its outgoing traffic utilizes the non-vpn default route. 
+
+For that, we will create an external binary that:
+- opens a socks5 proxy locally on a random port
+- generates a username and password for this session
+- prints the port, username, password to STDOUT so that the parent process may gather that info. 
+
+The VPN client will then pass that info to the Extension inside it's `status` response as an extra field. The extension must use that proxy for pages wanting to bypass the VPN.
+
+For Platforms where split tunneling is supported, the proxy is permanently fixed in the split-tunnel rule list, making sure the vpn connection cannot be used. 
+
+For Platforms where split tunneling is not available, the proxy will use socket options to make sure the right network device is used.
+> Q: do we need more details here? 
+
+If the daemon has support for per-app firewall rules (currently Windows), we will make sure only Firefox may access the port of the proxy to make sure other local apps cannot use the proxy to get the real IP of the user. 
+
+
+
+#### State 4: Device VPN is OFF & Page VPN is Off
+This State is trivial, no proxy needs to be set by the extension. 
+
+### Other API Changes 
+
+#### Renaming
+We will rename the current C++ Extension API classes to ... ExtensionAPI. They are currently under `server/` and called `ServerHandler` && `ServerConnection` which is an overloaded term for a VPN. 
+
+#### Split-Tunnel Lists
+Instead of reporting all excluded apps in `disabled_apps`, we will limit the response to executables containing `firefox`. The Extension may then limit it's funcitonality until resolved, as Firefox cannot access the VPN if it is excluded.  

```

Comment:
Using name convention is another attacking vector. A malicious app can duplicate itself, rename the new binary as 'firefox-foo' and workaround the VPN protection.

Reply:
Right now using the api has arbitrary read on `Settingsholder::excludedApps`. So any person with local code execution can learn of any excluded apps. 
An attacker may trivially forge a binary that would bypass this rule, (still a user would need to manually add foo-firefox)  but the attacker would already know that its own binary foo-firefox is on the system? Also it can trivially know it's split tunneled by introspecting it's created sockets. 
Can you elaborate what the attack vector is?